server/plugin.go

```diff
@@ -284,26 +288,42 @@ func (p *Plugin) disconnectGitHubAccount(userID string) {
 		return
 	}
 
-	p.API.KVDelete(userID + GITHUB_TOKEN_KEY)
-	p.API.KVDelete(userInfo.GitHubUsername + GITHUB_USERNAME_KEY)
+	if err := p.API.KVDelete(userID + githubTokenKey); err != nil {
+		p.API.LogWarn("Failed to delete github token", "userID", userID, "error", err.Error())
+	}
 
-	if user, err := p.API.GetUser(userID); err == nil && user.Props != nil && len(user.Props["git_user"]) > 0 {
-		delete(user.Props, "git_user")
-		p.API.UpdateUser(user)
+	if err := p.API.KVDelete(userInfo.GitHubUsername + githubUsernameKey); err != nil {
+		p.API.LogWarn("Failed to delete github token", "userID", userID, "error", err.Error())
+	}
+
+	user, err := p.API.GetUser(userID)
+	if err != nil {
+		p.API.LogWarn("Failed to get user props", "userID", userID, "error", err.Error())
+	} else {
+		_, ok := user.Props["git_user"]
+		if ok {
+			delete(user.Props, "git_user")
+			_, err := p.API.UpdateUser(user)
+			if err != nil {
+				p.API.LogWarn("Failed to get update user props", "userID", userID, "error", err.Error())
+			}
+		}
 	}
 
 	p.API.PublishWebSocketEvent(
-		WS_EVENT_DISCONNECT,
+		wsEventDisconnect,
 		nil,
 		&model.WebsocketBroadcast{UserId: userID},
 	)
 }
 
-func (p *Plugin) CreateBotDMPost(userID, message, postType string) *model.AppError {
+// CreateBotDMPost posts a direct message using the bot account.
+// Any error are not returned and instead logged.
+func (p *Plugin) CreateBotDMPost(userID, message, postType string) {
```

Comment:
Omitting the returned error and logging it instead just make the caller code simplere.